euphoria-core/src/main/java/cz/seznam/euphoria/core/client/io/DataSink.java

```diff
@@ -57,4 +58,19 @@ default void initialize() {}
    * @throws IOException if rolling back written out data fails for some reason
    */
   void rollback() throws IOException;
+
+
+  /**
+   * Called when adding output sink to the resulting DAG for processing.
+   * Purpose of this method is to enable sink to add arbitrary transformations
+   * to the dataset, before it gets persisted via this sink.
+   *
+   * @param output the dataset being written by this sink
+   * @return true if the flow was modified
+   */
+  default boolean onAdded(Dataset<T> output) {
```

Comment:
I guess we should find a better name for this :( Something like `prepareDataset`, `modifyFlow`...